src/ck-perf/trace-commSummary.C

```diff
@@ -0,0 +1,101 @@
+#include "charm++.h"
+#include "charm.h"
+#include "middle-conv.h"
+#include "trace-commSummary.h"
+#include "trace-commSummaryBOC.h"
+#include <fstream>
+
+CkpvStaticDeclare(TraceCommSummary*, _trace);
+
+CkGroupID traceCommSummaryGID;
+
+void _createTracecommSummary(char** argv)
+{
+  CkpvInitialize(TraceCommSummary*, _trace);
+  CkpvAccess(_trace) = new TraceCommSummary(argv);
+  CkpvAccess(_traces)->addTrace(CkpvAccess(_trace));
+}
+
+TraceCommSummary::TraceCommSummary(char** argv) : myPe(CkMyPe()), myNode(CkMyNode())
+{
+  if (CkpvAccess(traceOnPe) == 0) return;
+}
+
+void TraceCommSummary::beginExecute(CmiObjId* tid) { beginExecute(-1, -1, -1, myPe, 0); }
```

Comment:
Would it be possible to mark these `inline`?

Reply:
I assumed any competent compiler would take care of it. Do you think it would be beneficial?

Reply:
I think it is only guaranteed when member functions are defined in a class header, not in a cpp file: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-inline

Reply:
I'm not sure how much good it'll do here since they're invoked virtually via a `Trace*`, but I moved the definitions of the functions that just wrap the detailed `beginExecute` into the header.